### DISCUSSION TITLE
Serial.Read() error: -1

### DISCUSSION THREAD

**Discussion by rohoroho** [probonopd/MiniDexed#411] [Q&A]
Hi, I created the classic MIDI interface. I connected it between my Midi keyboard and the RPI -3.

When I hit a key, the monitor shows for every hit the following error:

serialmididevice: Serial.Read() error: -1

I checked the signal to the serial port with my scope. It looks fine. I have also tried to replave the resistor of 220K by a 100K resistor. But that did not help.

The baudrate in the configuration file I have not changed. 

The midi keyboard works fine with other programs.

What can be the problem?

Thanks

**Comment by BobanSpasic** [comment]
serialmididevice: Serial.Read() error: -1 means a BREAK is occurred.
A BREAK can occur mostly if there is no signal on the line for a long time (for the time of the 8 bits of transmission).
There can be a couple of reasons for something like this:
- voltage levels not matching (3V3 and 5V?)
- baud rate on Pi is set higher than on the transmitting device
- parity settings differs

I would check the voltage levels of the MIDI interface.

**Reply by rohoroho** [reply]
Thanks for your explanation! The solution Banana71 mentioned, I implemented and that worked!

**Comment by Banana71** [comment]
Hi rohoroho,

I would check the circuit for soldering errors or shorts.  Is the diode reverse biased?
Right RX Pin on GPIO ?
I have had good experience with the following circuit from @diyelectromusic:

![diy-midi-receive-3 3v_schem](https://user-images.githubusercontent.com/104296205/212185404-2df2add1-6593-4baa-8663-717dab9f2609.png)

**Reply by diyelectromusic** [reply]
> Hi rohoroho,
> 
> I would check the circuit for soldering errors or shorts. Is the diode reverse biased? Right RX Pin on GPIO ? I have had good experience with the following circuit from @diyelectromusic:
> 
> ![diy-midi-receive-3 3v_schem](https://user-images.githubusercontent.com/104296205/212185404-2df2add1-6593-4baa-8663-717dab9f2609.png)

These days I tend to use the H11L1 version with 3V3 systems like the Pi or Pico :)

Kevin

**Reply by rohoroho** [reply]
Great! I implemented and that worked! Thanks a lot!
